### ISSUE TITLE
why get_logalpha  is different from the paper,and how to understand logalpha < threshold

### ISSUE THREAD

**Issue by Zyazz** [zhuchen03/VIBNet#3]
`def get_logalpha(self):
        return self.post_z_logD.data - torch.log(self.post_z_mu.data.pow(2) + self.epsilon)
`
`def get_mask_hard(self, threshold=0):
        logalpha = self.get_logalpha()
        hard_mask = (logalpha < threshold).float()
        return hard_mask`

**Comment by daib13** [comment]
Hi @Zyazz , in the paper (arxiv version), we define \alpha in equation 11 on page 7, which is equivalent to the code here but with a subtle difference. In the paper, \alpha = \mu^2 / \sigma^2. However, the \alpha in the code is actually the inverse, which is \sigma^2/\mu^2. This is caused by a historical issue. We first define \alpha as what the code shows. Then we find it will be more consistent with some previous papers if we define \alpha as \mu^2 / \sigma^2. So we change the definition in the paper. But the code remains unchanged. Sorry for the confusion.

In the code, we set the mask to 1 if \log\alpha is smaller than the threshold. Because of the difference in the \alpha definition, we set the mask to 1 if \log\alpha is larger than the threshold in the paper.

**Comment by Zyazz** [comment]
It solve my problem, thanks!

**Comment by Zyazz** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    Hi ，I have some questions about the paper.
        
            In the paper, alpha is approximately equal to 0. In the code, alpha is greater than a threshold. How should the two be understood? In addition, how is this threshold selected and what is the reference basis.
        
    
    
        
            
        
    
    
     
    
        
                    
            
                
                    
                            
                                
                            
                            
                                Zachary
                            
                    
                        
                            
                                    xs
                                    邮箱：zy8920@gmail.com
                                    
                            
                        
                
            
        
        
    
    
 

    On 11/27/2019 11:56，Bin Dai<notifications@github.com> wrote： 


Hi @Zyazz , in the paper (arxiv version), we define \alpha in equation 11 on page 7, which is equivalent to the code here but with a subtle difference. In the paper, \alpha = \mu^2 / \sigma^2. However, the \alpha in the code is actually the inverse, which is \sigma^2/\mu^2. This is caused by a historical issue. We first define \alpha as what the code shows. Then we find it will be more consistent with some previous papers if we define \alpha as \mu^2 / \sigma^2. So we change the definition in the paper. But the code remains unchanged. Sorry for the confusion.
In the code, we set the mask to 1 if \log\alpha is smaller than the threshold. Because of the difference in the \alpha definition, we set the mask to 1 if \log\alpha is larger than the threshold in the paper.

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.
